stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -1525,6 +1525,33 @@ function findnz(S::AbstractSparseMatrixCSC{Tv,Ti}) where {Tv,Ti}
     return (I, J, V)
 end
 
+function Base.eachstoredindex(S::AbstractSparseMatrixCSC{Tv,Ti}) where {Tv,Ti}
+    numnz = nnz(S)
+    indices = Vector{CartesianIndex{2}}(undef, numnz)
+    count = 1
+    @inbounds for col = 1 : size(S, 2), k = getcolptr(S)[col] : (getcolptr(S)[col+1]-1)
```

Comment:
I'd be strongly in favor of replacing this complicated getcolptr-range simply by
```suggestion
    @inbounds for col = 1 : size(S, 2), k in nzrange(S, col)
```
more often.